src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -147,7 +148,13 @@ export class HttpConnection implements IConnection {
         }
 
         // Transport will not be null if state is connected
-        return this.transport!.send(data);
+        const send = () => this.transport!.send(data);
+        if (this.options.ensureMessageSendOrder) {
+            this.sendQueue = this.sendQueue.then(send, send);
```

Comment:
How should this work if the send fails?

Reply:
I think we should continue dequeuing after send failures. The error should be reported back to the caller though.